src/docs/guides/assets.md

```diff
@@ -0,0 +1,251 @@
+These types of files are referred to as 'assets', and include images, fonts, etc.
+
+In this guide, we describe different strategies for resolving assets on the filesystem.
+
+> CSS files are handled differently than assets; for more on using CSS, please see the [styling documentation](/docs/styling).
+
+## Asset Base Path
+
+The **asset base path** is the directory that Stencil will use to resolve assets.
+When a component uses an asset, the asset's location is resolved relative to the asset base path.
+
+The asset base path is automatically set for the following output targets:
+- [dist](/docs/distribution)
+- [hydrate](/docs/hydrate-app)
+- [www](/docs/www)
+
+For all other output targets, assets must be [moved](/docs/assets#manually-moving-assets) and the asset base path must be [manually set](/docs/assets#setassetpath).
+
+For each instance of the Stencil runtime that is loaded, there is a single asset base path.
+Oftentimes, this means there is only one asset base path per application using Stencil.
+
+## Resolution Overview
+
+The process of resolving an asset involves asking Stencil to build a path to the asset on the filesystem. 
+
+When an asset's path is built, the resolution is always done in a project's compiled output, not the directory containing the original source code.
+
+The example below uses the output of the [`www` output target](/docs/www) to demonstrate how assets are resolved.
+Although the example use the output of `www` builds, the general principle of how an asset is found holds for all output targets.
+
+When using the `www` output target, a `build/` directory is automatically created and set as the asset base path.
+An example `build/` directory and the assets it contains can be found below.
+
+```
+www/
+├── build/
+│   ├── assets/
+│   │   ├── logo.png
+│   │   └── scenery/
+│   │       ├── beach.png
+│   │       └── sunset.png
+│   └── other-assets/
+│       └── font.tiff
+└── ...
+```
+
+To resolve the path to an asset, Stencil's [`getAssetPath()` API](/docs/assets#getassetpath) may be used.
+When using `getAssetPath`, the assets in the directory structure above are resolved relative to `build/`.
+
+The code sample below demonstrates the return value of `getAssetPath` for different `path` arguments.
+The return value is a path that Stencil has built to retrieve the asset on the filesystem.
+```ts
+// with an asset base path of "/build/":
+
+// '/build/assets/logo.png'
+getAssetPath('assets/logo.png');
+// '/build/assets/scenery/beach.png'
+getAssetPath('assets/scenery/beach.png');
+// '/build/other-assets/font.tiff'
+getAssetPath('other-assets/font.tiff'); 
+```
+
+## Making Assets Available
+
+In order to be able to find assets at runtime, they need to be found on the filesystem from the output of a Stencil build.
+In other words, we need to ensure they exist in the distribution directory.
+This section describes how to make assets available under the asset base path.
+
+### assetsDirs
```

Comment:
Here's what I'm thinking. This section is probably the most practically important part of the docs as it shows how to actually use assets. Being so important, I wonder if this should be higher up in the doc so it's more easily available to those who just want to see the implementation quickly. I think the order you have right now makes perfect logical sense, but might not be the most practical.

Reply:
I'm not sure I agree - `assetsDirs` doesn't work for `dist-custom-elements` output target, and I'm a little concerned we'd be highlighting something that doesn't work for one of our default output targets.  

I think this option is important for sure, but in Stencil's current implementation of assets, the quick implementation is not something folks should be jumping to (IMO).  Assets are something that Stencil can't do a ton of legwork for on a developer's behalf.  Something like this is going to be for developers to work out themselves. I think this will especially true as they explore architectures that include one or more layers of component libraries

Reply:
Well defended. And I don't feel particularly passionate about this. Good to leave as is.